test/withStyles_test.jsx

```diff
@@ -142,6 +142,18 @@ describe('withStyles', () => {
       expect(StyledComponent.defaultProps).to.have.keys('foo');
     });
 
+    it('forwards refs through to the wrapped component', () => {
+      const stylesFn = sinon.stub().callsFake(() => ({}));
+      const MockComponent = React.forwardRef((props, forwardedRef) => (
+        <div ref={forwardedRef} />
+      ));
+      const StyledComponent = withStyles(stylesFn)(MockComponent);
+      const ref = React.createRef();
+      const wrapper = mountWithProviders(<StyledComponent ref={ref} />, providers);
+      const div = wrapper.find('div').first().getDOMNode();
+      expect(div).to.equal(ref.current);
+    });
+
```

Comment:
I didn't include an inverse of this test to assert that, for React<16.3, refs are _not_ forwarded because:

- I wasn't sure how to set up a test that runs for React<16.3
- I didn't think the test would be necessary if we plan on adding a breaking change to drop support for React<16.3 after this PR is merged